### PR TITLE
Preload fonts + real theme flicker fix + mixins for theme + deps updates + nvm update

### PR DESCRIPTION
**Pull request by colegoldsmith**
### Preloading fonts
Adding a head <link> to tell the browser to preload our fonts. This fixes the FOUT issue

### Real theme flicker fix
By adding the script which adds the `data-theme` to the html tag to the head section, it runs before the content is rendered and avoids the flicker.

### Mixins for theme
Instead of repeating the theme variables for the prefers theme and html[data-theme] we can use a mixin to write it once

### Deps updates
I found updating the postcss-custom-variables to work better, but required an npm upgrade. While doing so I upgraded the rest of the postcss deps

### Nvm update
Since one of our upgraded deps required an npm upgrade I changed the nvm to match the datastax-docs-site repo

### CODE REVIEW

File: gulp.d/tasks/build.js [hunk 1]
```diff
@@ -11,7 +11,8 @@ const ospath = require('path')
 const path = ospath.posix
 const postcss = require('gulp-postcss')
 const postcssCalc = require('postcss-calc')
-const postcssNesting = require('postcss-nesting')
+const postcssNested = require('postcss-nested')
+const postcssAdvancedVars = require('postcss-advanced-variables')
```
Comment (colegoldsmith) [inline gulp.d/tasks/build.js:15]: postcss-advanced-variables gives us mixins

File: gulp.d/tasks/build.js [hunk 2]
```diff
@@ -48,9 +49,9 @@ module.exports = (src, dest, preview) => () => {
         },
       },
     ]),
-    postcssNesting,
-    // NOTE because css vars are our theme switching mechanism we must preserve them with preserve: true
-    postcssVar({ preserve: true }),
+    postcssNested,
+    postcssAdvancedVars,
+    postcssVar({ preserve: preview }),
```
Comment (colegoldsmith) [inline gulp.d/tasks/build.js:54]: after upgrading postcss-custom-variables and npm this works properly now when preserve is false for the production bundle

File: package-lock.json
Comment (colegoldsmith) [inline package-lock.json:1]: Big package-lock update since I used node 16.20.2 to install

File: src/partials/head-scripts.hbs [hunk 3]
```diff
@@ -1,5 +1,24 @@
...
-    <script>function gtag(){dataLayer.push(arguments)};window.dataLayer=window.dataLayer||[];gtag('js',new Date());gtag('config','{{this}}')</script>
-    {{/with}}
-    {{!-- <script>var uiRootPath = '{{{uiRootPath}}}'</script> --}}
+{{#with site.keys.googleAnalytics}}
+<script async src="https://www.googletagmanager.com/gtag/js?id={{this}}"></script>
+<script>function gtag() { dataLayer.push(arguments) }; window.dataLayer = window.dataLayer || []; gtag('js', new Date()); gtag('config', '{{this}}')</script>
+{{/with}}
+{{!--
+<script>var uiRootPath = '{{{uiRootPath}}}'</script> --}}
+
```
Comment (colegoldsmith) [inline src/partials/head-scripts.hbs:7]: Preloading the fonts avoids the initial flicker when loading a new page. The icons were especially bad, and this fixes it.

File: src/partials/head-scripts.hbs [hunk 4]
```diff
@@ -1,5 +1,24 @@
...
+<link rel="preload" href="{{uiRootPath}}/font/material-icons-outlined-latin-400-normal.woff2" as="font"
+    type="font/woff2" crossorigin>
+<link rel="preload" href="{{uiRootPath}}/font/roboto-flex-latin-400-normal.woff2" as="font" type="font/woff2"
+    crossorigin>
+<link rel="preload" href="{{uiRootPath}}/font/roboto-mono-latin-500-normal.woff2" as="font" type="font/woff2"
+    crossorigin>
+
+<script>
```
Comment (colegoldsmith) [inline src/partials/head-scripts.hbs:15]: This script being in the head of the html adds the theme to the html element before content is loaded, thus avoiding the flicker in all cases

File: .nvmrc [hunk 5]
```diff
@@ -1 +1 @@
-10
+16.20.2
```
Comment (colegoldsmith) [inline .nvmrc:1]: After upgrading some packages, running with node 16 seems to be stable. Probably best to run with a more recent version and match version the datastax-docs-site repo uses

**Review by mlr — APPROVED** [review]

**Pull request merged**